### DISCUSSION TITLE
Remap keys for particular window class

### DISCUSSION THREAD

**Discussion by petrstepanov** [sezanzeb/input-remapper#233] [Ideas]
Hi! Are there any plans on implementing a filter for a key remap operation based on a current active window class? I am asking because this functionality is supported by the [Autokey](https://github.com/autokey/autokey) program. I was taking advantage of that. Please refer to the screenshot of the Autokey UI below:

<img src="https://i.imgur.com/txALNBr.png" />

On the bottom right there is a "Window Filter" option. The remap is applied only if current active window class matches the given REGEX expression. I do not know if this functionality is possible on Wayland? Just asking.

**Comment by sezanzeb** [comment]
Hey, this doesn't work in wayland unfortunately, see https://github.com/sezanzeb/key-mapper/discussions/20

I'd prefer to only have features in key-mapper that work independent of the environment. Someone would have to write a tool that works for the particular environment in question (X11, gnome, kde, etc.) if possible that uses key-mapper-control to achieve the desired effect https://github.com/sezanzeb/key-mapper/blob/main/readme/usage.md#cli

however, as of now activating a different preset freezes the input device for a second on my system, which will probably be super annoying for some third party tool. This might be improved soon due to https://github.com/sezanzeb/key-mapper/pull/228
